source/ui/Components/Output/SevenSegment/SevenSegmentDigit.qml

```diff
@@ -0,0 +1,320 @@
+
+/*
+ * C++ Assembler Interpreter
+ * Copyright (C) 2016 Chair of Computer Architecture
+ * at Technical University of Munich
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.0
+import QtQuick.Dialogs 1.0
+
+/*
+Represent one digit of a 7-segment display:
+   _
+  |_|
+  |_|
+
+*/
+
+Rectangle {
+    id: digitBackground
+
+    // Thickness of each segment
+    property var thickness: 20
+    // Background color for the rectangular region around the segments
+    property alias backgroundColor: digitBackground.color
+    // Color of the segments when active.
+    property color digitActiveColor: "yellow"
+    // Color of the segments when inactive.
+    property color digitInactiveColor: "black"
+    // Color of the text inside the segments
+    property color textColor: "gray"
+
+    // Determines whether the corresponding segment of the digit is activated.
+    property bool segment0Activated: false
```

Comment:
Property binding sadly does not work with values inside arrays. This means, using an array would require me to update model values manually, which is again difficult to accomplish in combination with the list model.